### ISSUE TITLE
Upgrading from 0.26.2 to 0.27 gives an exception and remains in a restart loop

### ISSUE THREAD

**Issue by alirezastack** [metabase/metabase#6495]
Current Metabase version: 0.26.2
OS: Debian 

When I want to upgrade my metabase to `version 0.27.0` it gives the error below and remains in a loop and shutdown then restart again:
```
11-29 06:46:25 INFO metabase.core :: Starting Metabase in STANDALONE mode
11-29 06:46:25 INFO metabase.core :: Launching Embedded Jetty Webserver with config:
 {:port 3000}

11-29 06:46:25 INFO metabase.core :: Starting Metabase version v0.27.0 (00e2d3e release-0.27.0) ...
11-29 06:46:25 INFO metabase.core :: System timezone is 'Etc/UTC' ...
11-29 06:46:26 WARN metabase.driver :: No -init-driver function found for 'metabase.driver.google'
11-29 06:46:27 INFO metabase.core :: Setting up and migrating Metabase DB. Please sit tight, this may take a minute...
11-29 06:46:27 INFO metabase.db :: Verifying h2 Database Connection ...
11-29 06:46:28 INFO metabase.db :: Verify Database Connection ...  ✅
11-29 06:46:28 INFO metabase.db :: Running Database Migrations...
11-29 06:46:28 INFO metabase.db :: Setting up Liquibase...
11-29 06:46:28 INFO metabase.db :: Liquibase is ready.
11-29 06:46:28 INFO metabase.db :: Checking if Database has unrun migrations...
11-29 06:46:31 INFO metabase.db :: Database Migrations Current ...  ✅
com.mchange.v2.cfg.DelayedLogItem [ level -> FINE, text -> "The configuration file for resource identifier 'hocon:/reference,/application,/c3p0,/' could not be found. Skipping.", exception -> null]
11-29 06:46:31 INFO db.migrations :: Running all necessary data migrations, this may take a minute.
11-29 06:46:31 INFO db.migrations :: Running data migration 'populate-card-database-id'...
org.h2.jdbc.JdbcSQLException: Referential integrity constraint violation: "FK_REPORT_CARD_REF_DATABASE_ID: PUBLIC.REPORT_CARD FOREIGN KEY(DATABASE_ID) REFERENCES PUBLIC.METABASE_DATABASE(ID) (-1337)"; SQL statement:
UPDATE "REPORT_CARD" SET "DATABASE_ID" = -1337 WHERE ("ID" in (361, 359, 360, 349, 351, 348)) [23506-194]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:345)
	at org.h2.message.DbException.get(DbException.java:179)
	at org.h2.message.DbException.get(DbException.java:155)
	at org.h2.constraint.ConstraintReferential.checkRowOwnTable(ConstraintReferential.java:371)
	at org.h2.constraint.ConstraintReferential.checkRow(ConstraintReferential.java:313)
	at org.h2.table.Table.fireConstraints(Table.java:976)
	at org.h2.table.Table.fireAfterRow(Table.java:994)
	at org.h2.command.dml.Update.update(Update.java:151)
	at org.h2.command.CommandContainer.update(CommandContainer.java:101)
	at org.h2.command.Command.executeUpdate(Command.java:258)
	at org.h2.jdbc.JdbcPreparedStatement.executeUpdateInternal(JdbcPreparedStatement.java:160)
	at org.h2.jdbc.JdbcPreparedStatement.executeUpdate(JdbcPreparedStatement.java:146)
	at com.mchange.v2.c3p0.impl.NewProxyPreparedStatement.executeUpdate(NewProxyPreparedStatement.java:105)
	at clojure.java.jdbc$db_do_execute_prepared_statement$fn__10886.invoke(jdbc.clj:901)
	at clojure.java.jdbc$db_transaction_STAR_.invokeStatic(jdbc.clj:714)
	at clojure.java.jdbc$db_transaction_STAR_.invoke(jdbc.clj:684)
	at clojure.java.jdbc$db_transaction_STAR_.invokeStatic(jdbc.clj:697)
	at clojure.java.jdbc$db_transaction_STAR_.invoke(jdbc.clj:684)
	at clojure.java.jdbc$db_do_execute_prepared_statement.invokeStatic(jdbc.clj:900)
	at clojure.java.jdbc$db_do_execute_prepared_statement.invoke(jdbc.clj:894)
	at clojure.java.jdbc$db_do_prepared.invokeStatic(jdbc.clj:932)
	at clojure.java.jdbc$db_do_prepared.invoke(jdbc.clj:912)
	at clojure.java.jdbc$execute_BANG_$execute_helper__10945.invoke(jdbc.clj:1161)
	at clojure.java.jdbc$execute_BANG_.invokeStatic(jdbc.clj:1165)
	at clojure.java.jdbc$execute_BANG_.invoke(jdbc.clj:1145)
	at toucan.db$execute_BANG_.invokeStatic(db.clj:324)
	at toucan.db$execute_BANG_.doInvoke(db.clj:319)
	at clojure.lang.RestFn.invoke(RestFn.java:410)
	at toucan.db$update_BANG_.invokeStatic(db.clj:378)
	at toucan.db$update_BANG_.invoke(db.clj:367)
	at toucan.db$update_where_BANG_.invokeStatic(db.clj:402)
	at toucan.db$update_where_BANG_.doInvoke(db.clj:392)
	at clojure.lang.RestFn.invoke(RestFn.java:464)
	at metabase.db.migrations$populate_card_database_id.invokeStatic(migrations.clj:335)
	at metabase.db.migrations$populate_card_database_id.invoke(migrations.clj:332)
	at metabase.db.migrations$run_migration_if_needed_BANG_.invokeStatic(migrations.clj:49)
	at metabase.db.migrations$run_migration_if_needed_BANG_.invoke(migrations.clj:40)
	at metabase.db.migrations$run_all_BANG_.invokeStatic(migrations.clj:68)
	at metabase.db.migrations$run_all_BANG_.invoke(migrations.clj:62)
	at clojure.lang.Var.invoke(Var.java:375)
	at metabase.db$run_data_migrations_BANG_.invokeStatic(db.clj:374)
	at metabase.db$run_data_migrations_BANG_.invoke(db.clj:369)
	at metabase.db$setup_db_BANG_.invokeStatic(db.clj:385)
	at metabase.db$setup_db_BANG_.doInvoke(db.clj:376)
	at clojure.lang.RestFn.invoke(RestFn.java:421)
	at metabase.core$init_BANG_.invokeStatic(core.clj:108)
	at metabase.core$init_BANG_.invoke(core.clj:87)
	at metabase.core$start_normally.invokeStatic(core.clj:194)
	at metabase.core$start_normally.invoke(core.clj:188)
	at metabase.core$_main.invokeStatic(core.clj:292)
	at metabase.core$_main.doInvoke(core.clj:287)
	at clojure.lang.RestFn.invoke(RestFn.java:397)
	at clojure.lang.AFn.applyToHelper(AFn.java:152)
	at clojure.lang.RestFn.applyTo(RestFn.java:132)
	at metabase.core.main(Unknown Source)
11-29 06:46:31 ERROR metabase.core :: Metabase Initialization FAILED:  Referential integrity constraint violation: "FK_REPORT_CARD_REF_DATABASE_ID: PUBLIC.REPORT_CARD FOREIGN KEY(DATABASE_ID) REFERENCES PUBLIC.METABASE_DATABASE(ID) (-1337)"; SQL statement:
UPDATE "REPORT_CARD" SET "DATABASE_ID" = -1337 WHERE ("ID" in (361, 359, 360, 349, 351, 348)) [23506-194]
11-29 06:46:31 INFO metabase.core :: Metabase Shutting Down ...
11-29 06:46:31 INFO metabase.core :: Metabase Shutdown COMPLETE
11-29 06:46:38 INFO metabase.util :: Loading Metabase...
11-29 06:46:43 INFO util.encryption :: DB details encryption is DISABLED for this Metabase instance. 🔓
See http://www.metabase.com/docs/latest/operations-guide/start.html#encrypting-your-database-connection-details-at-rest for more information.
11-29 06:46:48 INFO metabase.core :: Starting Metabase in STANDALONE mode
11-29 06:46:48 INFO metabase.core :: Launching Embedded Jetty Webserver with config:
 {:port 3000}

11-29 06:46:48 INFO metabase.core :: Starting Metabase version v0.27.0 (00e2d3e release-0.27.0) ...
11-29 06:46:48 INFO metabase.core :: System timezone is 'Etc/UTC' ...
11-29 06:46:49 WARN metabase.driver :: No -init-driver function found for 'metabase.driver.google'
11-29 06:46:50 INFO metabase.core :: Setting up and migrating Metabase DB. Please sit tight, this may take a minute...
11-29 06:46:50 INFO metabase.db :: Verifying h2 Database Connection ...
11-29 06:46:51 INFO metabase.db :: Verify Database Connection ...  ✅
11-29 06:46:51 INFO metabase.db :: Running Database Migrations...
11-29 06:46:51 INFO metabase.db :: Setting up Liquibase...
11-29 06:46:51 INFO metabase.db :: Liquibase is ready.
11-29 06:46:51 INFO metabase.db :: Checking if Database has unrun migrations...
11-29 06:46:54 INFO metabase.db :: Database Migrations Current ...  ✅
com.mchange.v2.cfg.DelayedLogItem [ level -> FINE, text -> "The configuration file for resource identifier 'hocon:/reference,/application,/c3p0,/' could not be found. Skipping.", exception -> null]
11-29 06:46:54 INFO db.migrations :: Running all necessary data migrations, this may take a minute.
11-29 06:46:54 INFO db.migrations :: Running data migration 'populate-card-database-id'...
org.h2.jdbc.JdbcSQLException: Referential integrity constraint violation: "FK_REPORT_CARD_REF_DATABASE_ID: PUBLIC.REPORT_CARD FOREIGN KEY(DATABASE_ID) REFERENCES PUBLIC.METABASE_DATABASE(ID) (-1337)"; SQL statement:
UPDATE "REPORT_CARD" SET "DATABASE_ID" = -1337 WHERE ("ID" in (361, 359, 360, 349, 351, 348)) [23506-194]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:345)
	at org.h2.message.DbException.get(DbException.java:179)
	at org.h2.message.DbException.get(DbException.java:155)
	at org.h2.constraint.ConstraintReferential.checkRowOwnTable(ConstraintReferential.java:371)
	at org.h2.constraint.ConstraintReferential.checkRow(ConstraintReferential.java:313)
	at org.h2.table.Table.fireConstraints(Table.java:976)
	at org.h2.table.Table.fireAfterRow(Table.java:994)
	at org.h2.command.dml.Update.update(Update.java:151)
	at org.h2.command.CommandContainer.update(CommandContainer.java:101)
	at org.h2.command.Command.executeUpdate(Command.java:258)
	at org.h2.jdbc.JdbcPreparedStatement.executeUpdateInternal(JdbcPreparedStatement.java:160)
	at org.h2.jdbc.JdbcPreparedStatement.executeUpdate(JdbcPreparedStatement.java:146)
	at com.mchange.v2.c3p0.impl.NewProxyPreparedStatement.executeUpdate(NewProxyPreparedStatement.java:105)
	at clojure.java.jdbc$db_do_execute_prepared_statement$fn__10886.invoke(jdbc.clj:901)
	at clojure.java.jdbc$db_transaction_STAR_.invokeStatic(jdbc.clj:714)
	at clojure.java.jdbc$db_transaction_STAR_.invoke(jdbc.clj:684)
	at clojure.java.jdbc$db_transaction_STAR_.invokeStatic(jdbc.clj:697)
	at clojure.java.jdbc$db_transaction_STAR_.invoke(jdbc.clj:684)
	at clojure.java.jdbc$db_do_execute_prepared_statement.invokeStatic(jdbc.clj:900)
	at clojure.java.jdbc$db_do_execute_prepared_statement.invoke(jdbc.clj:894)
	at clojure.java.jdbc$db_do_prepared.invokeStatic(jdbc.clj:932)
	at clojure.java.jdbc$db_do_prepared.invoke(jdbc.clj:912)
	at clojure.java.jdbc$execute_BANG_$execute_helper__10945.invoke(jdbc.clj:1161)
	at clojure.java.jdbc$execute_BANG_.invokeStatic(jdbc.clj:1165)
	at clojure.java.jdbc$execute_BANG_.invoke(jdbc.clj:1145)
	at toucan.db$execute_BANG_.invokeStatic(db.clj:324)
	at toucan.db$execute_BANG_.doInvoke(db.clj:319)
	at clojure.lang.RestFn.invoke(RestFn.java:410)
	at toucan.db$update_BANG_.invokeStatic(db.clj:378)
	at toucan.db$update_BANG_.invoke(db.clj:367)
	at toucan.db$update_where_BANG_.invokeStatic(db.clj:402)
	at toucan.db$update_where_BANG_.doInvoke(db.clj:392)
	at clojure.lang.RestFn.invoke(RestFn.java:464)
	at metabase.db.migrations$populate_card_database_id.invokeStatic(migrations.clj:335)
	at metabase.db.migrations$populate_card_database_id.invoke(migrations.clj:332)
	at metabase.db.migrations$run_migration_if_needed_BANG_.invokeStatic(migrations.clj:49)
	at metabase.db.migrations$run_migration_if_needed_BANG_.invoke(migrations.clj:40)
	at metabase.db.migrations$run_all_BANG_.invokeStatic(migrations.clj:68)
	at metabase.db.migrations$run_all_BANG_.invoke(migrations.clj:62)
	at clojure.lang.Var.invoke(Var.java:375)
	at metabase.db$run_data_migrations_BANG_.invokeStatic(db.clj:374)
	at metabase.db$run_data_migrations_BANG_.invoke(db.clj:369)
	at metabase.db$setup_db_BANG_.invokeStatic(db.clj:385)
	at metabase.db$setup_db_BANG_.doInvoke(db.clj:376)
	at clojure.lang.RestFn.invoke(RestFn.java:421)
	at metabase.core$init_BANG_.invokeStatic(core.clj:108)
	at metabase.core$init_BANG_.invoke(core.clj:87)
	at metabase.core$start_normally.invokeStatic(core.clj:194)
	at metabase.core$start_normally.invoke(core.clj:188)
	at metabase.core$_main.invokeStatic(core.clj:292)
	at metabase.core$_main.doInvoke(core.clj:287)
	at clojure.lang.RestFn.invoke(RestFn.java:397)
	at clojure.lang.AFn.applyToHelper(AFn.java:152)
	at clojure.lang.RestFn.applyTo(RestFn.java:132)
	at metabase.core.main(Unknown Source)
11-29 06:46:54 ERROR metabase.core :: Metabase Initialization FAILED:  Referential integrity constraint violation: "FK_REPORT_CARD_REF_DATABASE_ID: PUBLIC.REPORT_CARD FOREIGN KEY(DATABASE_ID) REFERENCES PUBLIC.METABASE_DATABASE(ID) (-1337)"; SQL statement:
UPDATE "REPORT_CARD" SET "DATABASE_ID" = -1337 WHERE ("ID" in (361, 359, 360, 349, 351, 348)) [23506-194]
11-29 06:46:54 INFO metabase.core :: Metabase Shutting Down ...
11-29 06:46:54 INFO metabase.core :: Metabase Shutdown COMPLETE
```

The error is specially about `Referential integrity constraint violation`. as you can see in the error above. 

Any help is appreciated.

**Comment by eebsidian** [comment]
I can confirm getting the same error.

**Comment by senior** [comment]
Thanks for reporting this issue! I have a fix for the problem in #6498. I'm going to close this issue as a duplicate of https://github.com/metabase/metabase/issues/6494. There is more info along with manual steps to fix it there.

**Comment by camsaul** [comment]
@alirezastack @eebsidian we've released a hotfix release, 0.27.1, which fixes this issue. You can download it here: https://www.metabase.com/start/
